### PR TITLE
refactor: switch to more foundry-native setup/tools

### LINKED ISSUE #663: [contract] refactor foundry setup
This issue aims to:

- [x] resolve the multiple `.gitmodules` issue that confuses `foundry`
- [x] utilize `forge fmt` (on top of solhint)
- [x] utilize `forge bind`, replacing original `gen-bindings.rs` scripts and workflow
- [x] update `contract-bindings/deploy.rs::{TestL1System, ...}` and `sequencer_utils::*` with more forge-native code (WIP)

cc @sveitser

### PR DESCRIPTION
**Pull request by alxiong**
Resolve #663 

(this PR is probably best reviewed commit-by-commit)

### CODE REVIEW

**Comment by nomaxg** [comment]
@alxiong is this ready for a look?

**Comment by alxiong** [comment]
Sorry, not yet. I'm trying to resolve the conflicts and update the previous `deploy.rs` APIs.

**Comment by alxiong** [comment]
Ready for review now @nomaxg @ImJeremyHe

**Review by nomaxg — APPROVED** [review]
LGTM, just one small question

File: .gitignore [hunk 1]
```diff
@@ -18,7 +18,6 @@ target/
 # vscode stuff
 .vscode
 
-# solidity files cache
-cache/
```
[the remark this answers is not in the document]
Comment (alxiong) [reply .gitignore:22]: git works recursively, there's `cache/` in `contracts/.gitignore`.

**Pull request merged**